compiler/rustc_data_structures/src/sync.rs

```diff
@@ -351,7 +351,7 @@ cfg_if! {
         use std::thread;
 
         #[inline]
-        pub fn join<A, B, RA: Send, RB: Send>(oper_a: A, oper_b: B) -> (RA, RB)
+        pub fn join<A, B, RA: std::marker::Send, RB: std::marker::Send>(oper_a: A, oper_b: B) -> (RA, RB)
```

Comment:
Can we wrap the results so these bounds can be `DynSend`?

Reply:
That makes sense!